src/config.rs

```diff
@@ -1315,22 +1392,126 @@ impl Default for CoprReadPoolConfig {
     }
 }
 
-#[derive(Clone, Serialize, Deserialize, PartialEq, Debug, Default)]
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct ReadPoolConfig {
+    pub use_yatp: bool,
+    pub yatp: YatpConfig,
     pub storage: StorageReadPoolConfig,
     pub coprocessor: CoprReadPoolConfig,
 }
 
 impl ReadPoolConfig {
-    pub fn validate(&mut self) -> Result<(), Box<dyn Error>> {
-        self.storage.validate()?;
-        self.coprocessor.validate()?;
+    pub fn validate(&self) -> Result<(), Box<dyn Error>> {
+        if self.use_yatp {
+            self.yatp.validate()?;
+        } else {
+            self.storage.validate()?;
+            self.coprocessor.validate()?;
+        }
         Ok(())
     }
 }
 
+impl Default for ReadPoolConfig {
+    fn default() -> ReadPoolConfig {
+        ReadPoolConfig {
+            use_yatp: false,
```

Comment:
Need to switch it to true in order to see how yatp affects Coprocessor performance by using our benchmark infra.